File: cpp_solutions/1791.cpp
class Solution {
public:
    int findCenter(vector<vector<int>>& edges) {
        int n = edges.size();
        vector<int> inorder(n + 2, 0);
        for (auto& edge : edges) {
            inorder[edge[0]] += 1;
            inorder[edge[1]] += 1;
        }
        for (int i = 1; i <= n + 1; ++i) {
            if (inorder[i] == n) return i;
        }
        return -1;
    }
};

File: rust_solutions/1791.rs
impl Solution {
    pub fn find_center(edges: Vec<Vec<i32>>) -> i32 {
        let n = edges.len();
        let mut indegrees = vec![0; n + 2];
        for i in 0..n {
            indegrees[edges[i][0] as usize] += 1;
            indegrees[edges[i][1] as usize] += 1;
        }
        for i in 1..n + 2 {
            if indegrees[i] == n {
                return i as i32;
            }
        }
        return -1;
    }
}

